### ISSUE TITLE
MicrocalDataSet.cut_summary isn't doing what it says it is.

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#144]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/144) by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Here's an example of the output of `ds.cut_summary()`. I removed most of the lines so we can focus on the important part, which is like this:

```text
In [361]: ds.cut_summary()
...
2974 pulses cut by both SAMECOL and POSTPEAK_DERIV
4299 pulses cut by both SAMECOL and CROSSTALK
0 pulses cut by both SAMECOL and SAMECOL
...
2974 pulses cut by POSTPEAK_DERIV
4299 pulses cut by CROSSTALK
0 pulses cut by SAMECOL
34565 pulses total

```

So wouldn't you agree that the numbers in the top 3 lines can only make sense if you replace the words "both X and Y" with the words "either X or Y"?

I don't know what you have in mind here. Did you want either/or, or both/and? In one case, the fix is to trivially change the text to read "either/or", but maybe *and* is what you wanted in the first place. If so, then you should change the logic of the method.

Separately, you should eliminate the line that gives "...cut by both X and X", because that's distracting and unnecessary.

Thanks to @pszypryt for pointing this out.

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

It was meant to be cut by both, though it seems unlikely I rely on that behavior very much. Given that it doesn't work.

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

Nevermind, I said something wrong.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Changed behavior of MicrocalDataSet.cut_summary().

Text now actually computes the AND and the OR of every pair of
cuts and reports both. Fixes #144.

Previous was computing the OR of the cuts but was calling it the AND.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Removing milestone: v0.7 (automated comment)
